src/services/services.ts

```diff
@@ -3059,30 +3060,22 @@ namespace ts {
                     }
                     return true;
                 }
-                else if (getAncestor(contextToken, SyntaxKind.JsxElement) || getAncestor(contextToken, SyntaxKind.JsxSelfClosingElement)) {
-                    // Go up until we hit either the element or expression
-                    let jsxNode = contextToken;
+                else if(jsxContainer) {
+                    let attrsType: Type;
+                    if (jsxContainer.kind === SyntaxKind.JsxSelfClosingElement) {
+                        // Cursor is inside a JSX self-closing element
+                        attrsType = typeChecker.getJsxElementAttributesType(<JsxSelfClosingElement>jsxContainer);
+                    }
+                    else if(jsxContainer.kind === SyntaxKind.JsxOpeningElement) {
+                        // Cursor is inside a JSX element
+                        attrsType = typeChecker.getJsxElementAttributesType(<JsxOpeningElement>jsxContainer);
+                    }
 
-                    while (jsxNode) {
-                        if (jsxNode.kind === SyntaxKind.JsxExpression) {
-                            // Defer to global completion if we're inside an {expression}
-                            break;
-                        } else if (jsxNode.kind === SyntaxKind.JsxSelfClosingElement || jsxNode.kind === SyntaxKind.JsxElement) {
-                            let attrsType: Type;
-                            if (jsxNode.kind === SyntaxKind.JsxSelfClosingElement) {
-                                // Cursor is inside a JSX self-closing element
-                                attrsType = typeChecker.getJsxElementAttributesType(<JsxSelfClosingElement>jsxNode);
-                            }
-                            else {
-                                Debug.assert(jsxNode.kind === SyntaxKind.JsxElement);
-                                // Cursor is inside a JSX element
-                                attrsType = typeChecker.getJsxElementAttributesType((<JsxElement>jsxNode).openingElement);
-                            }
-                            symbols = typeChecker.getPropertiesOfType(attrsType);
-                            isMemberCompletion = true;
-                            return true;
-                        }
-                        jsxNode = jsxNode.parent;
+                    if (attrsType) {
+                        symbols = typeChecker.getPropertiesOfType(attrsType);

```

Comment:
you probably should filter these based on the ones already defined. you do not want to keep suggesting the same property over and over. i am assuming it is expected to define the same property multiple time outside a spread.